### PR TITLE
Layers

### PR DESCRIPTION
**Pull request by shaunclarke2333**

### CODE REVIEW

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
- Please remove .terraform.lock.hcl - this file is not needed in git
- Pleaase commit level0/terraform.tfstate
- Please create level2 and move ec2 instance from level1 there

File: level2/backend.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Terraform backend configuration to store state file remotely in S3
+terraform {
+  backend "s3" {
+    bucket         = "main-backend-state-bucket"
+    key            = "mystate2/terraform.tfstate"
```
Comment (vladimirmukhin) [inline level2/backend.tf]: It's better for it to correspond to the name of the local folder i.e. level2/terraform.tfstate.
Same with level1

File: level2/variables.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# variable for main-public0 public subnet ID
+variable "subnetid" {
+  description = "variable for main-public0 public subnet ID"
+  default = "subnet-08056f0e3a471d580"
+}
+# variable for main-vpc ID
+variable "vpcid" {
+  description = "variable for main-vpc ID"
+  default = "vpc-073daf4e39f7b81ee"
+}
```
Comment (vladimirmukhin) [inline level2/variables.tf]: There is no need to have this as a variable. Instead, you can obtain required information directly from the state.
You can use the following construction for that: https://www.terraform.io/language/state/remote-state-data

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: level2/backend.tf (hunk 1 above)
Comment (shaunclarke2333) [reply level2/backend.tf]: Modfied backend to use matching remote state folders

File: level2/variables.tf (hunk 2 above)
Comment (shaunclarke2333) [reply level2/variables.tf]: Added terraform remote state data source

File: level2/variables.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  default     = "t2.small"
+}
+
+# key to connect to ec2
+variable "key_name" {
+  description = "key to connect to main-ec2 resource"
+  default     = "main"
+
+}
+
```
Comment (vladimirmukhin) [inline level2/variables.tf]: Please fix newlines

File: level1/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  value       = aws_vpc.my-main-vpc.id
+  sensitive   = true
+}
+
+# Output for main-public0 subnet id
+output "main-public0-subnet" {
+  description = "will be used by devices that depend on main-public0 subnet ID"
+  value       = aws_subnet.public_subnet[0].id
+  sensitive   = true
+}
```
Comment (vladimirmukhin) [inline level1/outputs.tf]: Please add trailing newline

**Review by vladimirmukhin — APPROVED** [review]

File: level1/outputs.tf (hunk 4 above)
Comment (shaunclarke2333) [reply level1/outputs.tf]: Added trailing line

File: level2/variables.tf (hunk 3 above)
Comment (shaunclarke2333) [reply level2/variables.tf]: removed new line below main.tf

**Pull request merged**